### PR TITLE
dev/core#852 Provide wysiwyg editor for confirmation email text

### PR DESCRIPTION
**Pull request by yashodha**
Overview
----------------------------------------
Provide wysiwyg editor for confirmation email text for online and offline event registrations
https://civicrm.stackexchange.com/questions/21255/confirmation-email-providing-a-wysiwyg-so-users-can-add-html-ified-content

Before
----------------------------------------
online
online_before

offline
offline_before

After
----------------------------------------
online
online_after

offline
offline_after

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@yashodha does this change how the text is saved (ie. add html tags to it). I feel like we tried something like this before & there was some gotchas

File: CRM/Event/Form/EventFees.php [hunk 1]
```diff
@@ -465,7 +465,7 @@ public static function buildQuickForm(&$form) {
 
     $form->add('select', 'from_email_address', ts('Receipt From'), $form->_fromEmails['from_email_id']);
 
-    $form->add('textarea', 'receipt_text', ts('Confirmation Message'));
+    $form->add('wysiwyg', 'receipt_text', ts('Confirmation Message'));
```
Comment (eileenmcnaughton) [inline CRM/Event/Form/EventFees.php]: if we do change we should change in the metadata & use addField method

**Comment by eileenmcnaughton** [comment]
I guess this is what I was thinking of https://issues.civicrm.org/jira/browse/CRM-13439 - the issue was that we can only have html OR text to go in both types of emails. If you could add a unit test to show it still comes out right in the text emails I think this is OK

@seamuslee001 does expanding wysiwyg have any security implications??

**Comment by seamuslee001** [comment]
Potentially yes, you would need to i think purify the output before letting it go into emails or such.

**Comment by prondubuisi** [comment]
Hello @eileenmcnaughton While trying to review this the credentials `pradmin` and `pradmin1234` don't seem to work 
[This is the review url](http://core-13976-8tqlp.test-1.civicrm.org:8001)

**Comment by colemanw** [comment]
@prondubuisi if you follow the [test sites](http://site-list.test-1.civicrm.org/?filter=core-13976-*) link from CiviBot's comment above it will tell you the correct login credentials.

**Comment by prondubuisi** [comment]
Hello @colemanw this works, is there any need to update the [review docs](https://docs.civicrm.org/dev/en/latest/core/pr-review/) to reflect this?

**Comment by eileenmcnaughton** [comment]
@prondubuisi yes please!

**Comment by prondubuisi** [comment]
> @prondubuisi yes please!

@eileenmcnaughton PR done https://github.com/civicrm/civicrm-dev-docs/pull/599

**Comment by eileenmcnaughton** [comment]
thanks @prondubuisi

**Comment by yashodha** [comment]
@eileenmcnaughton skipped _receipt_text_ for html input fields, _confirm_email_text_ was already there in the list

I need to check in text/html email but I don't think it changes how the text is saved. Since you could still do some rich text things but without the ease of editor.

**Comment by eileenmcnaughton** [comment]
@mlutfy you said you might be running something like this in prod & be able to 'vouch' for it

**Comment by mlutfy** [comment]
I also run this patch on my platform, however:

- when upgrading, people who had saved a text message with newlines will now get messages that are on a single line. In other words, this will be a regression for them? (they will lose some formatting)
- in the text version of the message template of the email, the text version should strip html?

**Comment by eileenmcnaughton** [comment]
We discussed in Pr review meeting & closing this as currently incomplete - would need 
- upgrade process to prevent regression
- some sort of converter routine in play for the html to text
- unit test

If you want to sort these things out please address & reopen -otherwise we will track in gitlab

**Pull request closed without merging**